hyperspy/io_plugins/tvips.py

```diff
@@ -0,0 +1,588 @@
+                raise ValueError(
+                    "Scan start and stop information could not be automatically "
+                    "determined. Please supply a scan_shape and scan_start_frame.")
+            total_scan_frames = record_idxs[scan_stop_frame]  # last rotator
+            scan_dim = int(np.sqrt(total_scan_frames))
+            if not np.allclose(scan_dim, np.sqrt(total_scan_frames)):
+                raise ValueError("Scan was not square, please supply a scan_shape and start_frame.")
+            scan_shape = (scan_dim, scan_dim)
+            # there may be discontinuities which must be filled up
+            indices = _guess_scan_index_grid(record_idxs, scan_start_frame, scan_stop_frame).reshape(scan_shape)
+        # scan shape and start are provided
+        else:
+            total_scan_frames = np.prod(scan_shape)
+            indices = np.arange(scan_start_frame, scan_start_frame + total_scan_frames).reshape(scan_shape)
+
+        # with winding scan, every second column or row must be inverted
+        # due to hysteresis there is also a predictable offset
+        if winding_scan_axis is not None:
+            if winding_scan_axis in ["x", 0]:
+                indices[..., ::2, :] = indices[..., ::2, :][..., :, ::-1]
+                indices[..., ::2, :] = np.roll(indices[..., ::2, :], hysteresis, axis=-1)
+            elif winding_scan_axis in ["y", 1]:
+                indices[..., :, ::2] = indices[..., :, ::2][..., ::-1, :]
+                indices[..., :, ::2] = np.roll(indices[..., :, ::2], hysteresis, axis=-2)
+            else:
+                raise ValueError("Invalid winding scan axis")
+
+        with dask.config.set(**{'array.slicing.split_large_chunks': True}):
+            data_stack = data_stack[indices.ravel()]
+        data_stack = data_stack.reshape(*indices.shape, dimy, dimx)
+        units = (indices.ndim - 2) * [""] + ['nm', 'nm', DPU, DPU]
+        names = (indices.ndim - 2) * [""] + ['y', 'x', 'dy', 'dx']
+        # no scale information stored in the scan!
+        scales = (indices.ndim - 2) * [1] + [1, 1, SDP, SDP]
+        offsets = (indices.ndim - 2) * [0] + [0, 0, offsety, offsetx]
+        # Create the axis objects for each axis
+        dim = data_stack.ndim
+        axes = [
+            {
+                'size': data_stack.shape[i],
+                'index_in_array': i,
+                'name': names[i],
+                'scale': scales[i],
+                'offset': offsets[i],
+                'units': units[i],
+                'navigate': True if i < len(scan_shape) else False,
+            }
+            for i in range(dim)]
+        if lazy:
+            chunking = {}
+            for ax_index in range(indices.ndim):
+                chunking[ax_index] = "auto"
+            for ax_index in range(indices.ndim, indices.ndim + 2):
+                chunking[ax_index] = None
+            data_stack = data_stack.rechunk(chunking)
```

Comment:
I've made `rechunking` a kwarg. If it's "falsy" no rechunking occurs. I'm pretty sure this would mean each file corresponds to a chunk. If it's "auto" = default then optimal rechunking occurs along navigation dimensions as above. If it's anything else, it's passed directly to the `chunks` argument in `dask.array.rechunk`, in case the user wants to do a custom rechunking.